cdap-tms/src/main/java/io/cdap/cdap/messaging/store/MessageTableKey.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.cdap.messaging.store;
+
+import io.cdap.cdap.api.common.Bytes;
+
+/**
+ * Row key used for an entry in a MessageTable.
+ */
+public class MessageTableKey {
+  private final byte[] rowKey;
+  private final int topicLength;
+  private long publishTimestamp;
+  private short sequenceId;
+
+  public MessageTableKey(byte[] topic) {
+    this.topicLength = topic.length;
+    rowKey = new byte[topicLength + Bytes.SIZEOF_LONG + Bytes.SIZEOF_SHORT];
+    Bytes.putBytes(rowKey, 0, topic, 0, topic.length);
+  }
+
+  public void set(long publishTimestamp, short sequenceId) {
+    this.publishTimestamp = publishTimestamp;
+    this.sequenceId = sequenceId;
+    Bytes.putLong(rowKey, topicLength, publishTimestamp);
+    Bytes.putShort(rowKey, topicLength + Bytes.SIZEOF_LONG, sequenceId);
+  }
+
+  public static MessageTableKey fromBytes(byte[] rowKey) {
```

Comment:
Move the static method to the top before the constructor to have better grouping.